org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/discovery/ZigBeeDiscoveryParticipant.java

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2010-2018 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.zigbee.discovery;
+
+import java.util.Map;
+import java.util.Set;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.config.discovery.DiscoveryResult;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.ThingUID;
+
+import com.zsmartsystems.zigbee.ZigBeeNode;
+
+/**
+ * A {@link ZigBeeDiscoveryParticipant} that is registered as a service is picked up by the ZigBeeDiscoveryService
+ * and can contribute {@link DiscoveryResult}s from ZigBee scans.
+ *
+ * @author Chris Jackson
+ *
+ */
+@NonNullByDefault
+public interface ZigBeeDiscoveryParticipant {
+
+    /**
+     * Defines the list of thing types that this participant can identify
+     *
+     * @return a set of thing type UIDs for which discovery results can be created
+     */
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs();
+
+    /**
+     * Returns the thing type UID for a ZigBee device
+     *
+     * @param node the {@link ZigBeeNode} found on the network
+     * @return a thing type UID or <code>null</code>, if the device is not supported by this participant
+     */
+    @Nullable
+    public ThingTypeUID getThingTypeUID(ZigBeeNode node);
```

Comment:
What is the reason that `getThingTypeUID()` is included in the interface? As far as I see, this method is not used by the `ZigBeeDiscoveryService` (but, in this PR, only internally in the `ZigBeeDefaultDiscoveryParticipant`).

Reply:
The interface was originally copied from the Bluetooth and originally I was using this method. However, if it's not used, we can remove it, but I prefer to make sure everyone is happy first ;)

Reply:
ok